File: Jerry.API/Models/Models/SaltTask.cs
using Jerry.API.Enums;

namespace Jerry.API.Models.Models;

public class SaltTask
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string SaltSelector { get; set; } = string.Empty;
    public SaltTaskStatus Status { get; set; }
    public int ProjectId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Project? Project { get; set; }
    public List<TaskUser> TaskUsers { get; set; } = [];
    public List<SaltCommand>? SaltCommands { get; set; } = [];

}

File: Jerry/Jerry.UI/src/app/interfaces/salt-task.ts
import { TaskStatus } from '../enums/enums';
import { Project } from './project';
import { SaltCommand } from './salt-command';
import { SaltTaskUser } from './salt-task-user';

export interface SaltTask {
  id: number;
  title: string;
  description: string;
  saltSelector: string;
  saltCommands: SaltCommand[];
  status: TaskStatus;
  project: Project;
  taskUsers: SaltTaskUser[];
  updatedAt: string;
  createdAt: string;
}
